File: showcase/starters/ag2/agent/tools/schedule_meeting.py
"""Schedule meeting tool implementation.

The HITL gating happens on the frontend via useHumanInTheLoop.
This tool just returns a pending approval status for the framework
wrapper to surface.
"""

from __future__ import annotations

from typing import Any

def schedule_meeting_impl(
    reason: str,
    duration_minutes: int = 30,
) -> dict[str, Any]:
    """Schedule a meeting (requires human approval).

    Returns a pending_approval status. The actual gating is done by the
    frontend's useHumanInTheLoop hook — the agent pauses until the user
    approves or rejects.
    """
    return {
        "status": "pending_approval",
        "reason": reason,
        "duration_minutes": duration_minutes,
        "message": (
            f"Meeting request: {reason} ({duration_minutes} min). "
            "Awaiting human approval."
        ),
    }

File: showcase/integrations/claude-sdk-typescript/shared-tools/schedule-meeting.ts
/**
 * Schedule meeting tool implementation — HITL gated.
 *
 * TypeScript equivalent of showcase/shared/python/tools/schedule_meeting.py.
 */

export interface ScheduleMeetingResult {
  status: "pending_approval";
  reason: string;
  duration_minutes: number;
  message: string;
}

export function scheduleMeetingImpl(
  reason: string,
  durationMinutes: number = 30,
): ScheduleMeetingResult {
  return {
    status: "pending_approval",
    reason,
    duration_minutes: durationMinutes,
    message: `Meeting request: ${reason} (${durationMinutes} min). Awaiting user time selection.`,
  };
}
